lua/sql.lua

```diff
@@ -235,17 +235,42 @@ end
 
 --- get sql table {name} schema, if table doesn't exist then return empty table.
 ---@param tbl string: the table name
----@param onlykeys boolean: whether to return a table of keys and their types. default false.
+---@param info boolean: whether to return table info. default false.
 ---@return table: list of keys or keys and their type.
-function sql:schema(tbl, onlykeys)
+function sql:schema(tbl, info)
   local tbl_sch = self:eval(string.format("pragma table_info(%s)",  tbl))
   if type(tbl_sch) == "boolean" then return {} end
-  if onlykeys then
-    local keys = {}
+  if info then
+    local tbl_info = {}
+    local req = {}
+    local def = {}
+    local types = {}
+
     for _, v in ipairs(tbl_sch) do
-      table.insert(keys, v.name)
+      tbl_info[v.name] = {
+        required = v.notnull == 1,
+        primary = v.ok == 1,
+        type = v.type,
+        cid = v.cid,
+        default = v.dflt_value
+      }
+    end
+
+    for k, v in pairs(tbl_info) do
+      if v.required then
+        req[k] = v
+      elseif v.default then
+        def[k] = v.default
+      end
+      types[k] = v.type
```

Comment:
Why not doing that the first time `tbl_sch` is iterated? You can just fill those tables as well? Can't we?

Reply:
indeed, yah I can merge them together, good catch